resources/CICD/GithubActions/deploy_AppStore.yml

```diff
@@ -0,0 +1,78 @@
+name: deploy_AppStore
+
+# SECRETS needed:
+### SSH_PRIVATE_KEY for Match Repo
+### MATCH_PASS
+### APPSTORE_CONNECT_API_KEY
+
+on:
+
+  push:
+    branches: [ master ]
+
+jobs:
+
+  Lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Run SwiftLint
+      uses: norio-nomura/action-swiftlint@3.1.0
+      with:
+        args: --strict
+        
+  Build:
```

Comment:
I think we need to separate this job into sub-jobs, for example:
- pre-action
- setup for build (bundle, pods)
- build tests
- set up ssh
- build and deploy
- download dsyms from app store
- upload dsyms to firebase

There are some missing steps, I think we need to bring this topic in the iOS chapter to discuss more to know what is the current ci jobs on different projects.

Reply:
I think the concept of Github Action `Jobs` is parallel actions running independently. It's possible to create sub jobs that wait for another but according to the [document](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds), I think it will look like a pyramid of doom.
[Android](https://github.com/nimblehq/android-templates/pull/76/files) approach is having the build workflow(with test) and another test that will run more frequently.

Reply:
:+1: So I think we need to update the steps into this `build` job :D

Reply:
In this case, I think we need to update about the steps of this `build` job :D